dspace-server-webapp/src/main/java/org/dspace/app/rest/WorkflowDefinitionController.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.model.CollectionRest;
+import org.dspace.app.rest.model.WorkflowDefinitionRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Collection;
+import org.dspace.core.Context;
+import org.dspace.xmlworkflow.factory.XmlWorkflowFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest controller that handles the config for workflow definitions
+ *
+ * @author Maria Verdonck (Atmire) on 11/12/2019
+ */
+@RestController
+@RequestMapping("/api/" + WorkflowDefinitionRest.CATEGORY + "/" + WorkflowDefinitionRest.NAME_PLURAL)
+public class WorkflowDefinitionController {
```

Comment:
This controller is misnamed & seems to be mapping the same request path as `WorkflowDefinitionRestRepository` (and `WorkflowDefinitionRest`).   I'd recommend renaming this to `WorkflowDefinitionCollectionController` (or similar), as this controller is specifically managing the Collection relationship here.  We can also then add a more specific `RequestMapping`...along the lines of:

```
@RequestMapping("/api/" + WorkflowDefinitionRest.CATEGORY + "/" + WorkflowDefinitionRest.NAME_PLURAL + "/{workflowName}/collections")
public class WorkflowDefinitionCollectionsController {
```

Reply:
Update: it's also worth considering whether this controller could be replaced with a `LinkRepository` to obtain the `collections` subresource.  That would better align this PR with the Projections work in #2625  (which has converted many similar Controller methods into LinkRepository classes)